tests/ops/op_math/test_controlled_decompositions.py

```diff
@@ -1090,7 +1097,115 @@ def f(bitstring):
         ).T
         assert np.allclose(u, np.eye(2 ** (n_ctrl_wires + 1)))
 
-    @pytest.mark.parametrize("n_ctrl_wires", range(3, 6))
+    @pytest.mark.parametrize("n_ctrl_wires", range(3, 10))
+    def test_decomposition_with_one_clean_worker_kg24(self, n_ctrl_wires):
+        """Test that the decomposed MCX gate using 1 clean ancilla performs the same unitary as the
+        matrix-based version by checking if U^dagger U applies the identity to each basis
+        state. This test focuses on the case where there is one work wire."""
+
+        # pylint: disable=protected-access
+        control_wires = Wires(range(n_ctrl_wires))
+        target_wire = n_ctrl_wires
+        work_wires = n_ctrl_wires + 1
+
+        dev = qml.device("default.qubit", wires=n_ctrl_wires + 2)
+
+        @qml.qnode(dev)
+        def f(bitstring):
+            qml.BasisState(bitstring, wires=range(n_ctrl_wires + 1))
+            qml.MultiControlledX(wires=list(control_wires) + [target_wire])
+            record_from_list(_decompose_mcx_with_one_worker_kg24)(
+                control_wires, target_wire, work_wires, work_wire_type="clean"
+            )
+            return qml.probs(wires=range(n_ctrl_wires + 1))
+
+        u = np.array(
+            [f(np.array(b)) for b in itertools.product(range(2), repeat=n_ctrl_wires + 1)]
+        ).T
+        assert np.allclose(u, np.eye(2 ** (n_ctrl_wires + 1)))
+
+    @pytest.mark.parametrize("n_ctrl_wires", range(3, 10))
+    def test_decomposition_with_one_dirty_worker_kg24(self, n_ctrl_wires):
```

Comment:
I don't think this sufficiently tests the correctness of the dirty implementation. Specifically because the "dirty qubit" has also started in the |0> state in this test.  You could either modify this test by preparing some random state in the work qubit, or write a separate unit test where you compare the input state to the output state of the circuit and show that only the target qubit was affected 👍🏼

Reply:
Doesn't `itertools.product([0, 1], n_ctrl_wires+1)` iterate through all possible bitstrings which would include non-zero state for dirty qubits?

Reply:
Ah okay, I see what you are trying to do. I think there might be a typo / bug because you actually have `n_ctrl_wires + 2` total wires. While you are preparing a bit string in the first `n_ctrl_wires + 1`, this actually excludes the work wire which is claimed to be dirty for this test.

Reply:
Fixed. This is what happens when you ctrl+c/v too much :laughing: